lib/retriable.rb

```diff
@@ -6,14 +6,22 @@
 module Retriable
   module_function
 
-  def self.configure
+  def configure
     yield(config)
   end
 
   def config
     @config ||= Config.new
   end
 
+  def retriable_with_context(context_key, options = {}, &block)
```

Comment:
I named it `with_context` initially, but the problem is I didn't like that if you use the kernel extension that you couldn't really put `with_context` on the Kernel since it's too generic. I didn't like the idea of having it be `#retriable` map to `Retriable.retriable` but `#retriable_with_context` in Kernel map to `Retriable.with_context`. 

I agree `retriable_with_context` is kind of verbose, I'm still thinking about it.